src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/cache_rotation.hpp

```diff
@@ -0,0 +1,198 @@
+
+    result_y = _mm512_fmadd_ps(cache_values_x, coeffts_sin, result_y);
+    result_y = _mm512_fmadd_ps(cache_values_y, coeffts_cos, result_y);
+
+    if (!is_underutilizing) {
+        mm512_uni_storeu_ps(current_x_values_ptr, result_x);
+        mm512_uni_storeu_ps(current_y_values_ptr, result_y);
+    } else {
+        mm512_uni_storeu_tail_ps(current_x_values_ptr, result_x, num_vectorized_elements_per_iteration);
+        mm512_uni_storeu_tail_ps(current_y_values_ptr, result_y, num_vectorized_elements_per_iteration);
+    }
+}
+#endif
+
+#if defined(HAVE_AVX2)
+template<class CT>
+inline static void rotate_kv_cache_chunk_avx2(CT* current_x_values_ptr, CT* current_y_values_ptr, float* current_rotation_coeffts_cos_ptr, float* current_rotation_coeffts_sin_ptr, size_t num_vectorized_elements_per_iteration, size_t is_underutilizing) {
+    using namespace ov::Extensions::Cpu::XARCH;
+
+    auto result_x = _mm256_setzero_ps();
+    auto result_y = _mm256_setzero_ps();
+
+    auto coeffts_cos = _mm256_undefined_ps();
+    auto coeffts_sin = _mm256_undefined_ps();
+
+    auto cache_values_x = _mm256_undefined_ps();
+    auto cache_values_y = _mm256_undefined_ps();
+
+    if (!is_underutilizing) {
+        coeffts_cos = mm256_uni_loadu_ps(current_rotation_coeffts_cos_ptr);
+        coeffts_sin = mm256_uni_loadu_ps(current_rotation_coeffts_sin_ptr);
+
+        cache_values_x = mm256_uni_loadu_ps(current_x_values_ptr);
+        cache_values_y = mm256_uni_loadu_ps(current_y_values_ptr);
+    }
+    else {
+        coeffts_cos = mm256_uni_loadu_tail_ps(current_rotation_coeffts_cos_ptr, num_vectorized_elements_per_iteration);
+        coeffts_sin = mm256_uni_loadu_tail_ps(current_rotation_coeffts_sin_ptr, num_vectorized_elements_per_iteration);
+
+        cache_values_x = mm256_uni_loadu_tail_ps(current_x_values_ptr, num_vectorized_elements_per_iteration);
+        cache_values_y = mm256_uni_loadu_tail_ps(current_y_values_ptr, num_vectorized_elements_per_iteration);
+    }
+
+    result_x = _mm256_fmadd_ps(cache_values_x, coeffts_cos, result_x);
+    result_x = _mm256_fnmadd_ps(cache_values_y, coeffts_sin, result_x); // negative multiply-add
+
+    result_y = _mm256_fmadd_ps(cache_values_x, coeffts_sin, result_y);
+    result_y = _mm256_fmadd_ps(cache_values_y, coeffts_cos, result_y);
+
+    if (!is_underutilizing) {
+        mm256_uni_storeu_ps(current_x_values_ptr, result_x);
+        mm256_uni_storeu_ps(current_y_values_ptr, result_y);
+    } else {
+        mm256_uni_storeu_tail_ps(current_x_values_ptr, result_x, num_vectorized_elements_per_iteration);
+        mm256_uni_storeu_tail_ps(current_y_values_ptr, result_y, num_vectorized_elements_per_iteration);
+    }
+}
+#endif
+
+template<class CT>
+inline static void rotate_kv_cache_block_hw(CT* cache_block_ptr, float* block_rotation_coefficients_ptr, size_t num_heads, size_t block_size, size_t embedding_size) {
+#if !defined(HAVE_AVX2) && !defined(HAVE_AVX512F)
```

Comment:
It should be cleaner if `rotate_kv_cache_block_hw` and `rotate_kv_cache_block_sw` are merged and let the `rotate_kv_cache_chunk_xxx` to handle the tails.

Reply:
I need HW and SW available as separate functions for testing purposes.

Reply:
The test style CPU plugin used is to use layer/subgraph([sdpa sample](https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/tests/functional/custom/subgraph_tests/src/classes/concat_sdp.hpp)) test to cover, due to CI infrastructure can cover avx2/avx512/amx, so there is no need to create a new test structure, we'd better to create a subgraph test just like sdpa.
But PagedAttention node is a little special, it's no reference now which means no reference result to compare, I think the reference will be coming. 
Base on current status, I suggest the new test module may be removed and file a ticket to record the work. 
Correct me if I'm wrong. @dmitry-gorokhov  @slyalin